docs/reference/projectyml.rst

```diff
@@ -100,72 +113,111 @@ example of a dev config:
 
-And set the following env variables:
+    The ``extra-index-url`` option will extend the extra indexes already set at the system level (if any).
 
-.. code-block:: bash
+    Other pip-related environment variables are not overridden.
 
-    export PIP_INDEX_URL=https://devpi.example.com/stable/
-    export PIP_PRE=false
+    To use the system's pip configuration without altering it, leave the corresponding option unset.
 
-In this scenario, pip options defined in env variables would be used over the system's pip config (if any).
 
 .. note::
 
-    Using netrc is the recommended way to set up authentication towards the index. See
-    this :ref:`section<setting_up_pip_index_authentication>` for more information.
+    Set up authentication towards the index using netrc. See this :ref:`section<setting_up_pip_index_authentication>`
+    for more information.
 
 .. _migrate_to_project_wide_pip_config:
 
 Migrate to project-wide pip config
 ----------------------------------
 
-Previously, there was no centralized way of configuring pip settings for the whole project. This section can be used
-as a migration guide.
+``inmanta-core 11.0.0`` introduced new options to configure pip settings for the whole project in a centralized way.
+For detailed information, see :ref:`here<specify_location_pip>`. The following code sample can be used as a baseline
+in the ``project.yml`` file:
+
+
+.. code-block:: yaml
+
+    pip:
+        index-url: https://my_repo.secure.company.com/repository/inmanta-production
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
+        pre: false
+        use-system-config: false
+
+When upgrading to ``inmanta-service-orchestrator 7.0.0`` or ``inmanta 2024.0``, this section can be used as
+a migration guide.
+
+
+All the v2 module sources currently set in a ``repo`` section of the ``project_yml`` with type ``package`` should
+also be duplicated in the ``pip.index-url`` (and ``pip.extra-index-url`` if more than one index is being used).
+
+If you want to allow pre-releases for v2 modules and other Python packages, set ``pip.pre = true`` in the project config
+file. This used to be controlled by the :class:`~inmanta.module.InstallMode` set at the project level or at a module
+level.
+
+Make sure the agents have access to the index(es) configured at the project level.
+
+Run a full compile after upgrading in order to export the project pip config to the server, so that it
+is available for agents. This will ensure that the agents follow the pip config defined in the project. For reference,
+prior to ``inmanta-core 11.0.0``, the agents were always using their respective system's pip config.
+
+Deprecation notes:
+""""""""""""""""""
+
+    - Allowing the installation of pre-release versions for v2 modules through the :class:`~inmanta.module.InstallMode`
+      is no longer supported. Use the project.yml ``pip.pre`` section instead.
+
+Breaking changes:
```

Comment:
I would also mention that `PIP_PRE` is now ignored unless `use-system-config` is set.